### PR TITLE
keymanager: api design

### PR DESCRIPTION
**Pull request by xenowits**
Introduces a new package `eth2util/keymanager` for keymanager api functionalities.

category: feature
ticket: https://github.com/ObolNetwork/charon/issues/1502

### CODE REVIEW

File: eth2util/keymanager/keymanager.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+// New returns a new Keymanager.
+func New(url string) Keymanager {
+	return Keymanager{
+		baseURL: url,
+	}
+}
+
+// Keymanager is the REST client for keymanager API requests.
+type Keymanager struct {
```
Comment (corverroos) [inline eth2util/keymanager/keymanager.go:42]: `Client`

File: eth2util/keymanager/keymanager.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	baseURL string // Base keymanager URL
+}
+
+// ImportKeystores pushes the keystores to keymanager.
+// See https://ethereum.github.io/keymanager-APIs/#/Local%20Key%20Manager/importKeystores.
+func (km Keymanager) ImportKeystores(ctx context.Context, keystores []keystore.Keystore, passwords []string) error {
```
Comment (corverroos) [inline eth2util/keymanager/keymanager.go:48]: verify len(keystores) == len(passwords)

**Review by corverroos — APPROVED** [review]
nice, add tests here as well please

File: eth2util/keymanager/keymanager.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+// New returns a new Client.
+func New(url string) Client {
+	return Client{
+		baseURL: url,
+	}
+}
+
+// Client is the REST client for keymanager API requests.
+type Client struct {
```
Comment (dB2510) [inline eth2util/keymanager/keymanager.go:42]: Why are we having a struct with only one field? why can we not simple pass baseURL as a function parameter and have pure functions?

Comment (xenowits) [reply eth2util/keymanager/keymanager.go:42]: We want to introduce a REST API client for [keymanager APIs](https://ethereum.github.io/keymanager-APIs/) just like we have go-eth2-client for ethereum beacon APIs.

**Pull request merged**